lib/App/index.js

```diff
@@ -3,35 +3,42 @@
 
 'use strict';
 
-const querystring = require('querystring');
+const {
+  join,
+  extname,
+  basename,
+  dirname,
+} = require('path');
+const { URLSearchParams } = require('url');
+const fs = require('fs');
+const util = require('util');
+
 const Ajv = require('ajv');
 const semver = require('semver');
 const tinycolor = require('tinycolor2');
+const imageSize = require('image-size');
 
 const Device = require('../Device');
 const Capability = require('../Capability');
 const Signal = require('../Signal');
 const Energy = require('../Energy');
 
+const statAsync = util.promisify(fs.stat);
+const readFileAsync = util.promisify(fs.readFile);
+const readDirAsync = util.promisify(fs.readdir);
+const lstatAsync = util.promisify(fs.lstat);
+const imageSizeAsync = util.promisify(imageSize);
+
 const {
-  openAsync,
-  readAsync,
-  statAsync,
-  readFileAsync,
-  readDirAsync,
-  lstatAsync,
-  imageSizeAsync,
-  join,
-  extname,
-  basename,
-  dirname,
+  readBytesFromFile,
 } = require('../../helpers');
 
-const VALIDATION_LEVELS = [
-  'debug',
-  'publish',
-  'verified',
-];
+const VALIDATION_LEVELS = {
```

Comment:
It's not per se implied that levels follow up on each other. It's better to manually specify them for every check, leaving room for future levels with different requirements.